src/core/instance/events.js

```diff
@@ -21,6 +22,8 @@ export function initEvents (vm: Component) {
 
 let target: Component
 
+const updateEventRE = /^update:([\w.$]+)\.([\w$]+)$/
```

Comment:
@jpsouzasilva Thanks for pointing this out, actually the event `'update:.foo.bar'` will not get into the Regex validation process unless there is an event named `'update:'` exists in the instance. However, we can modify the regex to `/^update:([\w$][\w.$]+)\.([\w$]+)$/` to make it more robust.